dbms/src/Parsers/ParserMoveQuery.cpp

```diff
@@ -0,0 +1,96 @@
+#include <Parsers/ASTIdentifier.h>
+#include <Parsers/ASTMoveQuery.h>
+
+#include <Parsers/CommonParsers.h>
+#include <Parsers/ParserMoveQuery.h>
+
+
+namespace DB
+{
+
+
+/// Parse database.table or table.
+static bool parseDatabaseAndTable(
+    ASTMoveQuery::Table & db_and_table, IParser::Pos & pos, Expected & expected)
+{
+    ParserIdentifier name_p;
+    ParserToken s_dot(TokenType::Dot);
+
+    ASTPtr database;
+    ASTPtr table;
+
+    if (!name_p.parse(pos, table, expected))
+        return false;
+
+    if (s_dot.ignore(pos, expected))
+    {
+        database = table;
+        if (!name_p.parse(pos, table, expected))
+            return false;
+    }
+
+    db_and_table.database.clear();
+    getIdentifierName(database, db_and_table.database);
+    getIdentifierName(table, db_and_table.table);
+
+    return true;
+}
+
+
+bool ParserMoveQuery::parseImpl(Pos & pos, ASTPtr & node, Expected & expected)
+{
+    ParserIdentifier name_p;
+    ParserKeyword s_table("TABLE");
+    ParserKeyword s_move_part("MOVE PART");
+    ParserKeyword s_to("TO");
+    ParserToken s_comma(TokenType::Comma);
```

Comment:
IMHO introducing completely separate command `TABLE xyz MOVE PART ... TO ... ` is bad idea.

It should be a part of ALTER TABLE, i.e. `ALTER TABLE xyz MOVE PART ... TO`.

Motivation:
1) all sql statements should start start from the verb. TABLE is not a verb. 
2) clickhouse already have ALTERs which (a) don't alter anything (b) executed one-replica only (w/o disturbing others) - see fetch partition / freeze partition.
2) other RDBMS also use ALTER statements for that. MS SQL: `ALTER DATABASE xxxx MODIFY FILE` , Oracle: `ALTER DATABASE RENAME FILE` etc.